VmAgent.Core/Interfaces/SystemOperations.cs

```diff
@@ -261,5 +300,50 @@ public void SetUnixFilePermissions(string filePath, int permissions)
             var unixFileInfo = new UnixFileInfo(filePath);
             unixFileInfo.FileAccessPermissions = (FileAccessPermissions)permissions;
         }
+
+        public void SetUnixOwnerIfNeeded(string path, bool applyToAllContents = false)
+        {
+            if (_vmConfiguration != null && _vmConfiguration.RunContainersInUsermode)
```

Comment:
is null allowed for vmConfiguration? Can we remove that check?

Reply:
This and the comment on the other PR got me thinking, I will remove SystemOperations.Default and change SystemOperations to Globals, both on Local MPS agent and in real Vm Agent. That's the clean way of dealing with this, the reason I put this in, is because I want Default to work and not crash out, and as a static initialized variable it can't have access to the VmConfig defined at runtime.

Reply:
If you don't object I vote we go with the ugly code in this PR then do the changes on Local  and VmAgent, which will be in a lot of places, which will make the PRs very dirty

Reply:
Yeah we can address that in a separate PR